core/modules/ccontrol/ParseRunner.cc

```diff
@@ -0,0 +1,155 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2013-2019 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+
+// Class header
+#include "ccontrol/ParseRunner.h"
+
+// Qserv headers
+#include "ccontrol/UserQuery.h"
+#include "parser/ParseException.h"
+#include "query/SelectStmt.h"
+#include "util/IterableFormatter.h"
+
+// these must be included before Log.h because they have a function called LOGS
+// that conflicts with the LOGS macro defined in Log.h
+#include "antlr4-runtime.h"
+#include "ccontrol/ParseListener.h"
+#include "parser/QSMySqlLexer.h"
+#include "parser/QSMySqlParser.h"
+
+// must come after QSMySqlLexer & Parser because of namespace collision
+#include "lsst/log/Log.h"
+
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.ccontrol.ParseRunner");
+
+
+class Antlr4ErrorStrategy : public antlr4::DefaultErrorStrategy {
+public:
+    explicit Antlr4ErrorStrategy(std::string const& statement) : _statement(statement) {}
+    Antlr4ErrorStrategy() = delete;
+    Antlr4ErrorStrategy(Antlr4ErrorStrategy const&) = delete;
+    Antlr4ErrorStrategy& operator=(Antlr4ErrorStrategy const&) = delete;
+
+private:
+    void recover(antlr4::Parser *recognizer, std::exception_ptr e) override {
+        LOGS(_log, LOG_LVL_ERROR, __FUNCTION__ <<
+                " antlr4 could not make a parse tree out of the input statement:" << _statement);
+        throw lsst::qserv::parser::ParseException(std::string("Failed to instantiate query: \"") + _statement +
+                std::string("\""));
```

Comment:
I don't think you need to wrap char* literals into std::string, `+` operator will do the right thing.